pkg/kube/secrets.go

```diff
@@ -154,12 +155,13 @@ func (r *secretsReader) ListByLocalObjectReferences(ctx context.Context, refs []
 	for _, secretRef := range refs {
 		var secret corev1.Secret
 		err := r.client.Get(ctx, client.ObjectKey{Name: secretRef.Name, Namespace: ns}, &secret)
-		if err != nil {
+		if err != nil && !k8sapierror.IsNotFound(err) {
```

Comment:
Wouldn't it be better to just `continue` if `k8sapierror.IsNotFound(err)`? Then you avoid protecting the `append`.